### DISCUSSION TITLE
[Loose idea - feel free to close at any moment in time] Provide some way to yield a "what's new?" in the gems ecosystem

### DISCUSSION THREAD

**Discussion by rubyFeedback** [rubygems/rubygems.org#2766] [Ideas]
Yo there ruby folks,

This is a loose idea I had, so it's not a good for an issue request - nonetheless let me explain
this, because perhaps the idea may have some merit in the long run.

Some time ago some entity added that diff-viewer where you can compare two different gems,
in particular seeing what changes were made specifically. This is, overall, I think useful - at
the least new functionality that wasn't available for many people prior to that. Anyway.

I had a very emails that my ruby requirements of 3.x ++ are too severe, so I downgraded 
most of the requirements. But afterwards I was wondering ... how to tell the users that
a change was made? I am most definitely never going to write any email because that
seems spammy/obstrusive/annoying. Yet how should they know otherwise? They can check
on their own, yes, but people don't have infinite time ... and people are different anyway.

So I was thinking it would be useful if rubygems itself could provide some sort of notification.
Hopefully in a non-intrusive way ... aka not spammy. Something that could be autogenerated
so that nobody has to invest time manually in that. A bit like discord (which I find spammy, by
default ... :P). Note that I don't suggest discord 1:1 as such; I am just giving this as an example
of news/notifications.

This can include security-related notifications too and use different colours (keep it simple though).

Now you can ask "but what kind of changes are newsworthy?". Good question. I am not necessarily
saying that only "newsworthy" changes are made available but more that ruby users and gem users
have some kind of way to see "trends" and "changes" in general - both in the whole gem ecosystem
(and ideally github too but I understand that this is too much work to focus on outside the
rubygems.org ecosystem), as well as (very important) the gems you are currently using. So, I think
the main idea is to focus on the gems that you use. This could be handled either by people having
an account on rubygems.org, or perhaps as opt-in for gem users (one has to check on this 
information not being abused by malicious actors, but I think that's possible; to keep this simple,
perhaps it is easier to only restrict this to those who have an account on rubygems.org by default,
and with a simple toggle option to specifically enable this in the setting before this is then shown).

The view-interface could be added to rubygems.org ... not sure about the API. Perhaps /news or
some additional statistical name. But anyway - these are details. I am more suggesting the idea.

The idea may be useful if you were not quite active in the last 2 months, but now want an update
and ideally want to ask rubygems.org rather than go through twitter and what not. Of course
this all depends on what is "newsworthy", but I suppose people can suggest what would be 
interesting for them. For me it may be interesting to see gems that were inactive, become
active again, or drop/change their requirements (if it is easy to see in a page that is ... that
is why I think an overview page like a HTML table or so may be useful in this regard).

Anyway - that's just a loose idea! Feel free to close this thread at any moment in time. Cheers!

**Comment by sonalkr132** [comment]
> I had a very emails that my ruby requirements of 3.x ++ are too severe, so I downgraded
most of the requirements.

these kinds of changes can be communicated through the changelog. if you add a link in spec.metadata.changelog_uri we will show it in the gem page sidebar. some gems already use after install hooks to show changelog and we already have bad feedback regarding it. you should also follow semvar to communicate breaking changes. I don't think we need more ways to do this.

> This can include security-related notifications too and use different colors (keep it simple though).

security-related notification is in the works.

Your tickets may be easier to read and considered more serious if you follow the issue template.
